### ISSUE TITLE
Support new svelte $$Generic type

### ISSUE THREAD

**Issue by SomaticIT** [sveltejs/eslint-plugin-svelte3#127]
Hello,

A couple of months ago (July 2021), the Svelte team added the ability to specify generics types when using TypeScript.

RFC: https://github.com/sveltejs/rfcs/pull/38
Thread: https://github.com/sveltejs/language-tools/issues/442

### Exemple

```svelte
<!-- ForEach.svelte -->
<script lang="ts">
  type T = $$Generic;
  export let items: T[];
</script>

{#each items as item}
  <slot {item} />
{/each}

<!-- App.svelte -->
<script lang="ts">
  import ForEach from "$lib/components/ForEach.svelte";
  import type { Todo } from "$lib/todo.service";

  let todos: Todo[] = [];
</script>

<ForEach items={todos} let:item>
  <!-- item is correctly typed as Todo here -->
  <p>{item.name}</p>
</ForEach>
```

### Issue

When using this new feature, `eslint-plugin-svelte3` complains with error `no-undef`.

### Workaround

Disabling the error manually:
```svelte
<script lang="ts">
  type T = $$Generic; /* eslint-disable-line no-undef */
  export let items: T[];
</script>

{#each items as item}
  <slot {item} />
{/each}
```

**Comment by Conduitry** [comment]
Your fix has been released as 3.2.1. Thank you!

**Comment by SomaticIT** [comment]
Happy to help! Thank you

**Comment by wvhulle** [comment]
So this feature has been removed in the successor of the parser. Is it a good idea to revert to this version while it is not reimplemented yet?

https://github.com/sveltejs/svelte-eslint-parser/issues/306
